[WPE] Log messages to the Android log where appropriate

aperezdc opened the pull request:
#### d52459e42f870cd1b42ad406f87071f981944b70
<pre>
[WPE] Log messages to the Android log where appropriate
<a href="https://bugs.webkit.org/show_bug.cgi?id=295155">https://bugs.webkit.org/show_bug.cgi?id=295155</a>

Reviewed by Nikolas Zimmermann and Sergio Villar Senin.

Add an Android-specific version of the logging macros. Those use
__android_log_print() to get the messages sent to the system log
facility. LogChannel is also augmented with calls to the Android
logging functions where appropriate.

Also, on actual devices Android closes the standard error stream of
processes by default, therefore it is handy to have messages send to
the standard error stream forwarded to the system log.

In all cases, the name of the LogChannel subsystem is set as the
Android logging tag, which means that the corresponding system
property (i.e. &quot;log.tag.&lt;name&gt;&quot;) needs to be set in order for logs
to be sent to the system log. This can be achieved for example with
the following command using ADB:

  adb shell setprop log.tag.WPEWebKit VERBOSE

Or, to save the setting:

  adb shell setprop persist.log.tag.WPEWebKit VERBOSE

The logging level may be set to VERBOSE, DEBUG, INFO, WARN, ERROR,
or FATAL. Those correspond to the log levels defined in &lt;android/log.h&gt;.
More information about how to configure the logger can be found at:

  <a href="https://android.googlesource.com/platform/system/core/+/66607ebc0e451/logd/README.property">https://android.googlesource.com/platform/system/core/+/66607ebc0e451/logd/README.property</a>

Setting environment variables on Android is a bit of a chore, so in
the future it may be desirable to read the list of enabled logging
channels from a system property instead.

* Source/JavaScriptCore/jsc.cpp:
(main):
* Source/WTF/wtf/Assertions.cpp:
* Source/WTF/wtf/Assertions.h:
* Source/WTF/wtf/Logger.h:
(WTF::Logger::willLog const):
(WTF::Logger::log):
(WTF::Logger::logVerbose):
* Source/WTF/wtf/PlatformWPE.cmake:
* Source/WTF/wtf/WTFConfig.h:
* Source/cmake/FindAndroid.cmake: Added.
* Source/cmake/OptionsWPE.cmake:

Canonical link: <a href="https://commits.webkit.org/297472@main">https://commits.webkit.org/297472@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fc7d865578c5c164e9793c28b92beecb8defb2e4

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/111897 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/31560 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/22037 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/117915 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/62125 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/32241 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/40141 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/85006 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/35694 "") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/114844 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/25746 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/100697 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/65446 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/25067 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/18832 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/61759 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/104385 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/95125 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/18903 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/121197 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/110456 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/38924 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/28947 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/93875 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/39305 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/96954 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/93699 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23911 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/38881 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/16669 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/34942 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/38822 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/44317 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/134717 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/38459 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/36264 "Found 1 new JSC stress test failure: wasm.yaml/wasm/stress/jitless-wasm-to-js-should-follow-to-wasm-call-frame-convention-2.js.default (failure)") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/41784 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/40174 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

spenap reviewed (commented):
Changes look good, this will be very useful :+1: !

nikolaszimmermann reviewed (approved):
rs=me, trusting you on the Android bits, where I have zero experience.

svillar commented on Source/WTF/wtf/Assertions.cpp:
```diff
@@ -144,6 +144,9 @@ static void logToStderr([[maybe_unused]] WTFLogChannel* channel, const char* buf
 #if PLATFORM(COCOA)
     os_log(channel ? channel->osLogChannel : webkitSubsystemForGenericOSLog(), "%s", buffer);
+#endif
+#if OS(ANDROID)
```
#elif ?

aperezdc replied:
Indeed, I will amend this before landing.

svillar reviewed (approved):
Just have a minor comment. Other than that r+

aperezdc committed:
[WPE] Log messages to the Android log where appropriate
https://bugs.webkit.org/show_bug.cgi?id=295155

Reviewed by Nikolas Zimmermann and Sergio Villar Senin.

Add an Android-specific version of the logging macros. Those use
__android_log_print() to get the messages sent to the system log
facility. LogChannel is also augmented with calls to the Android
logging functions where appropriate.

Also, on actual devices Android closes the standard error stream of
processes by default, therefore it is handy to have messages send to
the standard error stream forwarded to the system log.

In all cases, the name of the LogChannel subsystem is set as the
Android logging tag, which means that the corresponding system
property (i.e. "log.tag.<name>") needs to be set in order for logs
to be sent to the system log. This can be achieved for example with
the following command using ADB:

  adb shell setprop log.tag.WPEWebKit VERBOSE

Or, to save the setting:

  adb shell setprop persist.log.tag.WPEWebKit VERBOSE

The logging level may be set to VERBOSE, DEBUG, INFO, WARN, ERROR,
or FATAL. Those correspond to the log levels defined in <android/log.h>.
More information about how to configure the logger can be found at:

  https://android.googlesource.com/platform/system/core/+/66607ebc0e451/logd/README.property

Setting environment variables on Android is a bit of a chore, so in
the future it may be desirable to read the list of enabled logging
channels from a system property instead.

* Source/JavaScriptCore/jsc.cpp:
(main):
* Source/WTF/wtf/Assertions.cpp:
* Source/WTF/wtf/Assertions.h:
* Source/WTF/wtf/Logger.h:
(WTF::Logger::willLog const):
(WTF::Logger::log):
(WTF::Logger::logVerbose):
* Source/WTF/wtf/PlatformWPE.cmake:
* Source/WTF/wtf/WTFConfig.h:
* Source/cmake/FindAndroid.cmake: Added.
* Source/cmake/OptionsWPE.cmake:

Canonical link: https://commits.webkit.org/297472@main
